vnext/Microsoft.ReactNative/ReactApplication.h

```diff
@@ -4,15 +4,55 @@
 #pragma once
 
 #include "ReactApplication.g.h"
+#include <CppWinRTIncludes.h>
 #include "ReactNativeHost.h"
 
 namespace winrt::Microsoft::ReactNative::implementation {
 
-struct ReactApplication : ReactApplicationT<ReactApplication> {
-  using Super = ReactApplicationT<ReactApplication>;
+// NoDefaultCtorReactApplication_base is a copy of the generated ReactApplication_base
+// without the default constructor where it calls a factory for the base type.
+// This is done to fix the aggregation issue in types inheriting from the ReactApplication.
+// We call the factory in the ReactApplication constructor where we can pass correct
+// 'outer' interface.
+//
+// This class must match the generated ReactApplication_base.
+// It must be updated if the shape of generated ReactApplication_base is changed in future.
+// The only difference is that this class has no default constructor.
+template <typename D, typename... I>
+struct __declspec(empty_bases) NoDefaultCtorReactApplication_base
+    : implements<
+          D,
+          Microsoft::ReactNative::ReactApplication,
+          composable,
+          composing,
+          xaml::IApplicationOverrides,
+          xaml::IApplicationOverrides2,
+          I...>,
+      impl::require<D, xaml::IApplication, xaml::IApplication2, xaml::IApplication3>,
+      impl::base<D, xaml::Application>,
+      xaml::IApplicationOverridesT<D>,
+      xaml::IApplicationOverrides2T<D> {
+  using base_type = NoDefaultCtorReactApplication_base;
+  using class_type = Microsoft::ReactNative::ReactApplication;
+  using implements_type = typename NoDefaultCtorReactApplication_base::implements_type;
+  using implements_type::implements_type;
```

Comment:
>using implements_type::implements_type; [](start = 2, length = 39)

(nit) what does this do?

Reply:
It brings constructors from the implements_type base class. I am not sure why it is need. I effectively copy&pasted the generated code and did minimal changes to it: deleted default constructor.

---
In reply to: [418146298](https://github.com/microsoft/react-native-windows/pull/4755#discussion_r418146298) [](ancestors = 418146298)